diff2html/diff2html.d.ts

```diff
@@ -0,0 +1,66 @@
+// Type definitions for diff2html
+// Project: https://github.com/rtfpessoa/diff2html
+// Definitions by: rtfpessoa <https://github.com/rtfpessoa/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace Diff2Html {
+    
+    export interface Options {
+        inputFormat?: string;
+        outputFormat?: string;
+        showFiles?: boolean;
+        matching?: string;
+        synchronisedScroll?: boolean;
+        matchWordsThreshold?: number;
+        matchingMaxComparisons?: number;
+    }
+
+    export interface Line {
+        content: string;
+        type: string;
+        oldNumber: number;
+        newNumber: number;
+    }
+
+    export interface Block {
+        oldStartLine: number;
+        oldStartLine2?: number;
+        newStartLine: number;
+        header: string;
+        lines: Line[];
+    }
+
+    export interface Result {
+        addedLines: number;
+        deletedLines: number;
+        isCombined: boolean;
+        isGitDiff: boolean;
+        oldName: string;
+        newName: string;
+        language: string;
+        blocks: Block[];
+        oldMode?: string;
+        newMode?: string;
+        deletedFileMode?: string;
+        newFileMode?: string;
+        isDeleted?: boolean;
+        isNew?: boolean;
+        isCopy?: boolean;
+        isRename?: boolean;
+        unchangedPercentage?: number;
+        changedPercentage?: number;
+        checksumBefore?: string;
+        checksumAfter?: string;
+        mode?: string;
+    }
+    
+    export interface Diff2Html {
+        getJsonFromDiff(input: string, configuration?: Options): Result;
```

Comment:
Basically `export =` captures the namespace, value, and type meanings at any given location. You want to make some of those things you've defined accessible to others, so it'd be something like

```ts
namespace Diff2Html {
    export interface Options { /*...*/ }

    export functon getPrettyHtml(input: any, configuration?: Options): string;
}

declare module "diff2html" {
    export = Diff2Html;
}
```

Now your consumers can actually write

```ts
import d2h = require("diff2html");

let opts: d2h.Options = {
    // ...
}
```